assets/src/block-editor/plugins/amp-preview-button.js

```diff
@@ -105,22 +105,49 @@ function writeInterstitialMessage( targetDocument ) {
 /**
  * A 'Preview AMP' button, forked from the Core 'Preview' button: <PostPreviewButton>.
  *
- * Rendered into the DOM with renderPreviewButton() in helpers/index.js.
- * This also moves the (non-AMP) 'Preview' button to before this, if it's not already there.
+ * Inserted into the DOM immediately after the 'Preview' button.
  *
  * @see https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/editor/src/components/post-preview-button/index.js#L95-L200
  */
-class AMPPreview extends Component {
+class AmpPreviewButton extends Component {
 	/**
 	 * Constructs the class.
 	 *
 	 * @param {*} args Constructor arguments.
 	 */
 	constructor( ...args ) {
 		super( ...args );
-		this.moveButton = this.moveButton.bind( this );
-		this.openPreviewWindow = this.openPreviewWindow.bind( this );
+
 		this.buttonRef = createRef();
+		this.openPreviewWindow = this.openPreviewWindow.bind( this );
+
+		this.root = document.createElement( 'div' );
+		this.root.id = 'amp-wrapper-post-preview';
```

Comment:
Maybe this should be a class instead of an ID. While unlikely, it's technically possible for this component to be rendered multiple times on a page.